### PR TITLE
Mark the lookup join tests in IndexResolverFieldNamesTests as snapshot-only

### LINKED ISSUE #118810: [CI] JdbcDocCsvSpecIT test {docs.testFilterToday} failing
**Build Scans:**
- [elasticsearch-periodic-platform-support #5412 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/qh6g724bgfq2i)
- [elasticsearch-periodic-platform-support #5412 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/q27nlfowrt4po)
- [elasticsearch-pull-request #47033 / part-3](https://gradle-enterprise.elastic.co/s/mfh2rdhmzbl2q)
- [elasticsearch-periodic-platform-support #5412 / opensuse-leap-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/clxtdmkry6iik)
- [elasticsearch-periodic-platform-support #5412 / almalinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/vesuoxeisjvym)
- [elasticsearch-pull-request #47032 / part-3](https://gradle-enterprise.elastic.co/s/2igbikwrbuudy)
- [elasticsearch-pull-request #47018 / part-3](https://gradle-enterprise.elastic.co/s/por4hvvbpa46a)
- [elasticsearch-pull-request #47028 / part-3](https://gradle-enterprise.elastic.co/s/nlmtmkryt7sg6)
- [elasticsearch-pull-request #47027 / part-3](https://gradle-enterprise.elastic.co/s/ehyzmiknpfqv4)
- [elasticsearch-pull-request #47024 / part-3](https://gradle-enterprise.elastic.co/s/ijlfwtqtu53he)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:sql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.sql.qa.single_node.JdbcDocCsvSpecIT" -Dtests.method="test {docs.testFilterToday}" -Dtests.seed=9C8EAE0BA171F958 -Dtests.locale=en-TV -Dtests.timezone=Antarctica/Syowa -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.sql.qa.single_node.JdbcDocCsvSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bdocs.testFilterToday%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.junit.ComparisonFailure: Different result for column [first_name], entry [5] expected:<B[oja]n> but was:<B[rendo]n>
```

**Issue Reasons:**
- [main] 7 consecutive failures in test test {docs.testFilterToday}
- [main] 2 consecutive failures in step almalinux-8_platform-support-unix
- [main] 3 consecutive failures in step part-3
- [main] 2 consecutive failures in step opensuse-leap-15_platform-support-unix
- [main] 9 consecutive failures in step part3
- [main] 2 consecutive failures in step windows-2022_checkpart3_platform-support-windows
- [main] 2 consecutive failures in step amazonlinux-2023_platform-support-aws
- [main] 2 consecutive failures in step almalinux-8-aarch64_checkpart3_platform-support-arm
- [main] 2 consecutive failures in step ubuntu-2004-aarch64_checkpart3_platform-support-arm
- [main] 2 consecutive failures in step openjdk21_checkpart3_java-fips-matrix
- [main] 2 consecutive failures in step openjdk23_checkpart3_java-matrix
- [main] 2 consecutive failures in step openjdk21_checkpart3_java-matrix
- [main] 2 consecutive failures in pipeline elasticsearch-periodic-platform-support
- [main] 3 consecutive failures in pipeline elasticsearch-pull-request
- [main] 9 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### LINKED ISSUE #118814: [CI] IndexResolverFieldNamesTests class failing
**Build Scans:**
- [elasticsearch-periodic #5427 / release-tests](https://gradle-enterprise.elastic.co/s/6mlswxudzfdie)
- [elasticsearch-periodic #5422 / release-tests](https://gradle-enterprise.elastic.co/s/5jghzck5ecrgy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.session.IndexResolverFieldNamesTests.testMultiLookupJoinKeepAfter" -Dtests.seed=FAE91D4D4D82D6AE -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=my-Mymr-MM -Dtests.timezone=Pacific/Pitcairn -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.session.IndexResolverFieldNamesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.esql.session.IndexResolverFieldNamesTests),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.elasticsearch.xpack.esql.parser.ParsingException: line 3:3: mismatched input 'LOOKUP' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where'}
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step release-tests
- [8.x] 2 failures in class org.elasticsearch.xpack.esql.session.IndexResolverFieldNamesTests (1.3% fail rate in 155 executions)
- [8.x] 2 failures in step release-tests (66.7% fail rate in 3 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by astefan**
Fixes https://github.com/elastic/elasticsearch/issues/118814
Fixes https://github.com/elastic/elasticsearch/issues/118810

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

**Comment by carlosdelest** [comment]
This is also failing in 8.17 (see [this PR](https://github.com/elastic/elasticsearch/pull/118818)), tagging backport for that version

**Review by alex-spies — APPROVED** [review]
LGTM, thanks @astefan !

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 1]
```diff
@@ -1364,6 +1365,7 @@ public void testMetrics() {
     }
 
     public void testLookupJoin() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1368]: Surely the capability should be JOIN_LOOKUP_V6 (or even higher if more work gets in before this PR).

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 2]
```diff
@@ -1372,6 +1374,7 @@ public void testLookupJoin() {
     }
 
     public void testLookupJoinKeep() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1377]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java (hunk 1 above)
Comment (craigtaverner) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1368]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 3]
```diff
@@ -1385,6 +1388,7 @@ public void testLookupJoinKeep() {
     }
 
     public void testLookupJoinKeepWildcard() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1391]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 4]
```diff
@@ -1398,6 +1402,7 @@ public void testLookupJoinKeepWildcard() {
     }
 
     public void testMultiLookupJoin() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1405]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 5]
```diff
@@ -1410,6 +1415,7 @@ public void testMultiLookupJoin() {
     }
 
     public void testMultiLookupJoinKeepBefore() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1418]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 6]
```diff
@@ -1423,6 +1429,7 @@ public void testMultiLookupJoinKeepBefore() {
     }
 
     public void testMultiLookupJoinKeepBetween() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1432]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 7]
```diff
@@ -1447,6 +1454,7 @@ public void testMultiLookupJoinKeepBetween() {
     }
 
     public void testMultiLookupJoinKeepAfter() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1457]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 8]
```diff
@@ -1473,6 +1481,7 @@ public void testMultiLookupJoinKeepAfter() {
     }
 
     public void testMultiLookupJoinKeepAfterWildcard() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1484]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 9]
```diff
@@ -1486,6 +1495,7 @@ public void testMultiLookupJoinKeepAfterWildcard() {
     }
 
     public void testMultiLookupJoinSameIndex() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1498]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 10]
```diff
@@ -1499,6 +1509,7 @@ public void testMultiLookupJoinSameIndex() {
     }
 
     public void testMultiLookupJoinSameIndexKeepBefore() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1512]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 11]
```diff
@@ -1513,6 +1524,7 @@ public void testMultiLookupJoinSameIndexKeepBefore() {
     }
 
     public void testMultiLookupJoinSameIndexKeepBetween() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1527]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java [hunk 12]
```diff
@@ -1538,6 +1550,7 @@ public void testMultiLookupJoinSameIndexKeepBetween() {
     }
 
     public void testMultiLookupJoinSameIndexKeepAfter() {
+        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.LOOKUP_V4.isEnabled());
```
Comment (craigtaverner) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java:1553]:
```suggestion
        assumeTrue("LOOKUP JOIN available as snapshot only", EsqlCapabilities.Cap.JOIN_LOOKUP_V6.isEnabled());
```

**Review by craigtaverner — CHANGES_REQUESTED** [review]
Seems like the wrong capability was used. These queries use LOOKUP JOIN, so sufrely the JOIN_LOOKUP_V6 capability should be the one to use?

**Comment by alex-spies** [comment]
@craigtaverner this uses the correct capability now, and should be fine to be merged :)

**Pull request merged**